docs/new_api.md

```diff
@@ -0,0 +1,646 @@
+
+For example, in `GET /api/visitors`, the response body is:
+```
+{ result: [{ id, name, gender, email, ... }] }
+```
+We should specify a filter on `gender` and `email` in the following way:
+```
+GET /visitors?gender=foo&email=bar
+```
+
+In addition, the following query parameters should be standard:
+* `fields`: Comma separated list of fields in the response body to return. For example, `GET /visitors?fields=name,gender` returns only the `name` and `gender` field in the response body, instead of the entire object.
+* `sort`: Single field by which to sort results
+* `order`: Order by which to sort field specified by `sort`. One of `asc` or `desc`
+* `since`: ISO86001 date string (with timezone?) specifying the **earliest** time from which results should be returned. This is not always meaningful for every endpoint, and is therefore ignored in those cases.
+* `until`: ISO86001 date string (with timezone?) specifying the **latest** time from which results should be returned. This is not always meaningful for every endpoint, and is therefore ignored in those cases.
+
+Note: Specifying the query via the request body is preferred where possible. Query parameters are mostly intended for use by `GET` requests.
+
+
+
+## Standard Request Body
+```
+{
+  "query": {},
+  "sort"?: { field: "asc" | "desc", ... },
+  "filter"?: {}
+}
+```
+
+## Standard Response Body
+```
+{
+  "result"?: {} | [],
+  "meta"?: {},
+  "error"?: { "message": String, "validation"?: [String] }
+}
+```
+
+## Error Response
+All non-successful responses should return appropriate status codes in the range `4xx` or `5xx`, with `"result": null`, and a descriptive message in `"error": { "message": "..." }`. If the error is due to validation, the `"error"` object should have a key of `"validation"` which is an array of strings describing the validation errors.
+
+## Accept and Content-Type Headers
+Clients should set the `Accept` header in order to indicate what content-type(s) they expect as a response. If no `Accept` header is set, the API will default to a `application/json` content-type. The API will set the `Content-Type` header accurately to reflect the response format. The two response formats that should be supported are:
+* JSON
+* XLS (via XML)
+
+## Authentication and Authorization
+Clients are authenticated on the API via a JWT, which is passed through the `Authorization` header. Permissions are role-based, not fine-grain. The roles a client has are identified by the JWT signature. There are four roles, listed below.
+
+* `NULL`: Un-authenticated user
+* `CB`: Community business user
+* `CB_ADMIN`: Community business user with elevated admin permissions
+* `TWINE_ADMIN`: Twine platform admin user
+
+The `CB_ADMIN` permission level is a short-lived authorization with broader rights than the `CB` permission level. It is attained via a permission escalation mechanism (currently confirming password).
+
+## Activities
+### `GET    /api/activities/`
```

Comment:
I'm not clear on this layout. Is this meant to be a request to this endpoint without params by a CB?

Reply:
It's meant to be a request to that endpoint by a client that has a token corresponding to a `CB` role.